File: Appliquer/parrot.rb
class Pet

  def initialize(name, greeting="Hello", legs=4, cri="Ouaf")
    @name = name
    @greeting = greeting
    @legs = legs
    @cri = cri
  end

  def say_hi()
    puts ("#{@greeting}, I'm #{@name}, #{@cri}")
  end

  def count_legs()
    puts(@legs)
  end

end

class Cat < Pet
  def initialize(name)
    super(name, greeting="Hello", legs=4, cri="miaouu")
  end
end

class Parrot < Pet
  def initialize(name)
    super(name, greeting="Coucou", legs=2, cri ="ouii")
  end
end

pitch = Pet.new("Pitch")
felix = Cat.new("Felix")
kevin = Parrot.new("Kevin")
pitch.say_hi
felix.say_hi
pitch.count_legs
kevin.say_hi
kevin.count_legs

File: Appliquer/parrot.py
class Pet:
    legs = 0
    def __init__(self, name ="Boulbi", greeting = "Hello", legs = 4, cri = "ouaf"):
        self.name = name
        self.greeting = greeting
        self.legs = legs
        self.cri = cri

    def say_hi(self):
        print(f"{self.greeting}, I'm {self.name}!")

    @classmethod
    def legs_count(cls):
        return cls.legs




class Cat(Pet):
	def __init__(self, name):
		super().__init__("cat", "Meow", 4, "miaaa")


class Parrot(Pet):
    def __init__(self, name):
        super().__init__("par", "Coucou", 2, "oui")


    def say_hi(self):
        print(f"{self.greeting}, My name is {self.name}!")
        print(self.legs)



jack = Parrot("Jack")

jack.say_hi()
jack.legs_count()
Pet.legs_count()
print(Cat.legs_count())
print(Parrot.legs_count())
